lib/ie-contract.js

```diff
@@ -22,5 +23,16 @@ export const createMeridianContract = async (contractAddress = SparkImpactEvalua
     SparkImpactEvaluator.ABI,
     provider
   )
+
+  try {
+    await pTimeout(ieContract.currentRoundIndex(), {
+      milliseconds: 10_000
```

Comment:
should I increase this to 20s or 60s

Reply:
Don't want to give up too soon in case a backend is just slow but overall available

Reply:
another alternative: Start using FallbackProvider. Wdyt @bajtos?

Reply:
> This surfaces another issue: What if the backend becomes unavailable later on? Would be better to make the process fail then as well

💯 

I think it will be more often that the RPC backend becomes unavailable while the service is running, rather than at the service startup.

> another alternative: Start using FallbackProvider. Wdyt @bajtos?

IIRC, we have had a good experience with FallbackProvider in one of the other Spark services. I think it's a great idea to use it in spark-evaluate too.

Reply:
> should I increase this to 20s or 60s?
>
> Don't want to give up too soon in case a backend is just slow but overall available

I think 10 seconds is a good start; we can easily tweak this value later based on what we learn about real-world performance.

Have you considered what happens when none of the providers can respond in 10 seconds? Our service will always exit after ~10 seconds; Fly.io will restart it then we exit again, etc.

I think it would be great to create a Papertrail->Slack alert when this happens for more than N minutes, where N can be like 5-10 minutes.

@juliangruber WDYT?

Reply:
> Have you considered what happens when none of the providers can respond in 10 seconds? Our service will always exit after ~10 seconds; Fly.io will restart it then we exit again, etc.

This papertrail search should handle it:

> "spark-evaluate errors" search found 5 matches in the past ten minutes 

Do you think it's worth the extra effort?

Reply:
Related: https://github.com/filecoin-station/spark-evaluate/issues/312

Reply:
I'm adding https://github.com/filecoin-station/spark-evaluate/issues/281 to the current milestone

Reply:
> > Have you considered what happens when none of the providers can respond in 10 seconds? Our service will always exit after ~10 seconds; Fly.io will restart it then we exit again, etc.
> 
> This papertrail search should handle it:
> 
> > "spark-evaluate errors" search found 5 matches in the past ten minutes
> 
> Do you think it's worth the extra effort?

I think we want an alert to let us know when this is happening, don't we? Is there any other alert that would be eventually triggered when spark-evaluate keeps crashing at startup? (E.g. no round evaluation scores submitted or no new RSR entries created.) If yes, then that's good enough for me.

Reply:
Well, isn't that the alert that already exists?